### ISSUE TITLE
Chef Client 13 is skipping datadog::repository

### ISSUE THREAD

**Issue by andreimahu** [DataDog/chef-datadog#457]
**Chef Client Version:** 13.x.x
**Issue:** When calling datadog::dd-agent in a role runlist it skips the datadog::repository recipe entirely. The run fails afterwards at the below resource because the prerequisites did not happen:

```
directory node['datadog']['config_dir'] do
  if is_windows
    owner 'Administrators'
    rights :full_control, 'Administrators'
    inherits false
  else
    owner 'dd-agent'
    group 'root'
    mode '755'
  end
end
```

**Platform:** Amazon Linux AMI 2017.03.1 (HVM), SSD Volume Type - ami-d7b9a2b1

**Mentions:** 
- The node['datadog']['installrepo'] attribute is left as default which is true.
- It runs fine with chef 12.x.x
- Running chef-client with the why-run option did not produce any useful output. The "thought" process between 12 and 13 is the same. Chef client 13 simply ignores the recipe for some reason.
- I attached 2 screenshots to compare the run process. You can see exactly the resources skipped by chef 13. After passing the "datadog-api-key-unset", it jumps straight to: ( 6 resources skipped )

```
package 'datadog-agent-base' do
    action :remove
    only_if 'rpm -q datadog-agent-base' if %w(rhel fedora).include?(node['platform_family'])
    not_if 'apt-cache policy datadog-agent-base | grep "Installed: (none)"' if node['platform_family'] == 'debian'
end
```

![chef-client12-run](https://user-images.githubusercontent.com/16101257/28366175-d7d2f1a0-6c8b-11e7-8ff0-455cbd034ebe.PNG)
![chef-client13-run](https://user-images.githubusercontent.com/16101257/28366177-db2c5864-6c8b-11e7-9928-8dc8b0b0cee8.PNG)

To reproduce the behaviour running datadog::dd-agent with any chef-client 13 should do it.

**Comment by andreimahu** [comment]
I managed to find what the issues is. Amazon Linux instances now have their own platform family defined with Chef 13 by ohai. This cookbook has no case for node['platform_family'] = 'amazon' and this is why most of the stuff does not run.

**Comment by olivielpeau** [comment]
Was fixed in `2.12.0` through #448, forgot to close this issue at the time.

Thanks @andreimahu for reporting this
